internal/alpini/router/alpini-router-base/src/main/java/com/linkedin/alpini/router/ScatterGatherRequestHandlerImpl.java

```diff
@@ -932,27 +933,32 @@ protected CompletionStage<?> appendErrorForEveryKey(
       appendError(request, responses, status, contentMsg.append(", RoutingPolicy=").append(roles).toString(), ex);
     } else {
       // For requests with keys, send an error for each key. TODO: Consider if we could rip all of that out?
-      for (K key: part.getPartitionKeys()) {
-        complete = complete.thenApply(aVoid -> pathParser.substitutePartitionKey(basePath, key))
-            .thenCompose(pathForThisKey -> {
-              contentMsg.append(", PartitionName=")
-                  .append(_scatterGatherHelper.findPartitionName(pathForThisKey.getResourceName(), key));
-              return null;
-            })
-            .exceptionally(e -> {
-              LOG.info("Exception in appendErrorForEveryKey, key={}", key, e);
-              return null;
-            })
-            .thenApply(aVoid -> {
-              appendError(
-                  request,
-                  responses,
-                  status,
-                  contentMsg.append(", RoutingPolicy=").append(roles).toString(),
-                  ex);
-              return null;
-            });
-      }
+      complete = complete.thenCompose(aVoid -> {
+        List<CompletableFuture<String>> list = part.getPartitionKeys()
+            .stream()
```

Comment:
Can we please change to a non-streaming code? We want to avoid it in the hotpath due to the high amount of garbage it generates

Reply:
redone without `stream()`